web_widget_slick/static/src/js/web_widget_slick.js

```diff
@@ -0,0 +1,172 @@
+/* Copyright 2016-2017 LasLabs Inc.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html). */
+
+odoo.define('web_widget_slick', function(require) {
+    "use strict";
+    var core = require('web.core');
+    var relational_fields = require('web.relational_fields');
+    var registry = require('web.field_registry');
+
+    var FieldOne2Many = relational_fields.FieldOne2Many;
+
+    var FieldSlickImages = FieldOne2Many.extend({
+
+    className: 'o_slick',
+    template: 'FieldSlickImages',
+    $slick: null,
+    no_rerender: true,
+    loading: [],
+    loaded: 0,
+
+    events: {
+        'mousedown img': function(ev) {
+          ev.preventDefault();
+        },
+        'touchstart img': function(ev) {
+          ev.preventDefault();
+        },
+        // Triggering a resize on the lazyLoaded event prevents the carousel
+        // from appearing empty when page loads
+        'lazyLoaded': function(ev) {
+          $(ev.target).trigger('resize');
+        }
+    },
+
+    defaults: {
+      lazyLoad: 'ondemand',
+      fieldName: 'datas',
+      modelName: 'ir.attachment',
+      slidesToShow: 3,
+      slidesToScroll: 1,
+      swipeToSlide: true,
+      dots: true,
+      infinite: true,
+      speed: 500,
+      arrows: true,
+      responsive: [
+        {
+          breakpoint: 1024,
+          settings: {
+            slidesToShow: 2,
+            slidesToScroll: 1
+          }
+        },
+        {
+          breakpoint: 600,
+          settings: {
+            slidesToShow: 1,
+            slidesToScroll: 1
+          }
+        }
+        // You can unslick at a given breakpoint now by adding:
+        // settings: "unslick"
+        // instead of a settings object
+      ]
+    },
+
+    init: function (parent, name, record, options) {
+        this._super.apply(this, arguments);
+        this.nodeOptions = _.defaults(this.nodeOptions, this.defaults);
+    },
+
+    destroy: function () {
+        if (this.$slick) {
+            var $imgs = this.$el.find('img');
+            // Unslicking removes the carousel but re-appends any images,
+            // so removal of images is also required
+            $imgs.each($.proxy(this._slickRemove, this));
+            this.$slick.slick('unslick');
+          }
+        this._super.apply(this, arguments);
+    },
+
+    _render: function () {
+        this._super();
+        this.$el.empty()
+        this.$slick = $('<div class="slick-container"></div>');
+        if (this.nodeOptions.arrows) {
+            this.$slick.addClass('slick-arrowed');
+        }
+        this.$el.append(this.$slick);
```

Comment:
instead of flushing and then appending later (flickering might happen) can't you do `this.$el.html(this.$slick)`?